src/slsV4BucketPolicies.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  S3Client,
+  ListBucketsCommand,
+  GetBucketPolicyCommand,
+  PutBucketPolicyCommand,
+} from "@aws-sdk/client-s3";
+
+const s3Client = new S3Client({ region: "us-east-1" });
+
+async function getServerlessBuckets(): Promise<string[]> {
+  const { Buckets } = await s3Client.send(new ListBucketsCommand({}));
+  return (Buckets || [])
+    .map((bucket) => bucket.Name!)
+    .filter((name) => name.startsWith("serverless-framework-"));
+}
+
+async function getBucketPolicy(bucketName: string): Promise<string | null> {
+  try {
+    const { Policy } = await s3Client.send(
+      new GetBucketPolicyCommand({ Bucket: bucketName })
+    );
+    return Policy || null;
+  } catch (error: any) {
+    if (error.name === "NoSuchBucketPolicy") {
+      return null;
+    }
+    throw error;
+  }
+}
+
+async function enforceHttpsPolicy(bucketName: string) {
+  const policy = JSON.stringify({
+    Version: "2012-10-17",
+    Statement: [
+      {
+        Sid: "AllowSSLRequestsOnly",
+        Action: "s3:*",
+        Effect: "Deny",
+        Resource: [
+          `arn:aws:s3:::${bucketName}`,
+          `arn:aws:s3:::${bucketName}/*`,
```

Comment:
Since these buckets are used by lots of different branches deploys, we probably are going to have to be looser than a specific bucket here. I'm concerned this would make it work just for the last person who's deployed.

Reply:
I'm confused.  Can we huddle?

Reply:
I misunderstood, looks great.